core/state_processor.go

```diff
@@ -221,17 +223,19 @@ type UtxosCreatedDeleted struct {
 // transactions failed to execute due to insufficient gas it will return an error.
 func (p *StateProcessor) Process(block *types.WorkObject, batch ethdb.Batch) (types.Receipts, []*types.Transaction, []*types.Log, *state.StateDB, uint64, uint64, uint64, *multiset.MultiSet, []common.Unlock, error) {
 	var (
-		receipts     types.Receipts
-		usedGas      = new(uint64)
-		usedState    = new(uint64)
-		header       = types.CopyWorkObject(block)
-		blockHash    = block.Hash()
-		nodeLocation = p.hc.NodeLocation()
-		nodeCtx      = p.hc.NodeCtx()
-		blockNumber  = block.Number(nodeCtx)
-		parentHash   = block.ParentHash(nodeCtx)
-		allLogs      []*types.Log
-		gp           = new(types.GasPool).AddGas(block.GasLimit())
+		receipts                              types.Receipts
+		usedGas                               = new(uint64)
+		usedState                             = new(uint64)
+		header                                = types.CopyWorkObject(block)
+		blockHash                             = block.Hash()
+		nodeLocation                          = p.hc.NodeLocation()
+		nodeCtx                               = p.hc.NodeCtx()
+		blockNumber                           = block.Number(nodeCtx)
+		parentHash                            = block.ParentHash(nodeCtx)
+		allLogs                               []*types.Log
+		gp                                    = new(types.GasPool).AddGas(block.GasLimit())
+		numTxsProcessed                       = big.NewInt(0)
+		blockMinFee, blockMaxFee, blockAvgFee *big.Int
```

Comment:
no need to initialize these variables to zero?

Reply:
They're set upon first use in the function so that we don't get confused thinking min or max = 0.